### ISSUE TITLE
Puyo Puyo Tetris (546050)

### ISSUE THREAD

**Issue by novadragonDOTspace** [ValveSoftware/Proton#359]
SFX and voices play, but the Music doesn't 

Using Manjaro Linux latest packages on a Lenovo g50-70 laptop

**Comment by arzardk** [comment]
You need to attach game log file and your system specs via [github gist](https://gist.github.com) to this issue. You can obtain log file if you running Proton with user settings, which you can find in your /home directory. Path to this folder is something like this `/home/arzardk/.steam/steam/steamapps/common/Proton 3.7`. In this folder you find file called `user_settings.example.py` or something similar. Just rename it to `user_settings.py` and run your game again. You can find log file in your /home directory after that.

**Comment by novadragonDOTspace** [comment]
Hardwareinfo compiled by hardinfo:
https://gist.github.com/feuerpanda/3b5a1132ed865f0038aa0cc6ebf92baa
PPT Log: https://gist.github.com/feuerpanda/dc7ad84600935da6c6feaab114c87ee7

Am Fr., 24. Aug. 2018 um 12:05 Uhr schrieb Viktor <notifications@github.com

**Comment by kisak-valve** [comment]
Puyo Puyo™Tetris® audio issue

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1215.
@Inglebard posted on 2018-09-03T18:50:11:

Hi,

I just test Puyo Puyo™Tetris® on proton 3.7-3.

The game seems to work globally.
There is audio issue. Maybe more information here : https://appdb.winehq.org/objectManager.php?sClass=version&iId=36583 .

Tested on : Inspiron 15 3567
OS : Archlinux
CPU : intel i5-7200u
GPU : intel i5-7200u
RAM : 8G

**Comment by novadragonDOTspace** [comment]
It seems like WineHQ Bugzilla already has a bug open for this and a solution in there https://bugs.winehq.org/show_bug.cgi?id=44938

**Comment by novadragonDOTspace** [comment]
Should still be looked upon, behaviour seems to completly mirror the WineHQ Bug

**Comment by HonkingGoose** [comment]
@flibitijibibo Is this something you want labeled "XAudio2" maybe?

Relevant quotes from [bugreport by user morningpee from WineHQ Bugzilla](https://bugs.winehq.org/show_bug.cgi?id=44938), (I've bolded the relevant parts):
>I include the "freezing" bit because I think it is related. Whenever the game freezes, the terminal that I launched the program from shows this line:
00a7:err:ntdll:RtlpWaitForCriticalSection section 0x8438a8 "../../../wine-staging/dlls/xaudio2_7/xaudio_dll.c: **IXAudio2Impl.lock**" wait timed out in thread 00a7, blocked by 00b6, retrying (60 sec)

>The addresses are unpredictable, of course. The second time it happened, I got the same error but with different addresses.
0082:err:ntdll:RtlpWaitForCriticalSection section 0x34dc7270 "../../../wine-staging/dlls/xaudio2_7/xaudio_dll.c: **XA2SourceImpl.lock**" wait timed out in thread 0082, blocked by 0092, retrying (60 sec)

>The problem seems related to https://bugs.winehq.org/show_bug.cgi?id=39517 , but  the proposed fix does not work for me. That is, I tried installing Microsoft's **xaudio** implementation

>winetricks --force **xact**

>but the problem persists after doing this (and restarting the game). Downgrading to wine-staging 3.4-1 makes the problem go away.

**Comment by flibitijibibo** [comment]
Looks like XAudio2 to me!

**Comment by jolty1** [comment]
**Only tested on proton 3.16-4 beta!**

I got the music to work and fix crashing without **winetricks --force xact** by editing the games user.reg in steamapps/compatdata/546050/pfx/ 

Add new line under

[Software\\Wine\\DllOverrides]

`"xaudio2_7"="native,builtin"`

**Comment by Inglebard** [comment]
Hi,

I use `winetricks --force xact`, the missing sounds works now, but sometimes I have a kind of "echo", like sounds play twice or more.

I also notice input issue if I connect 2 controllers (Steam controller +  ps3 dualshock sixaxis ). It's hard to explain but it's like if the input are mixed and sometimes not detected.

**Comment by aljelly** [comment]
A while ago I got this game working before Proton was launched, using the NVIDIA proprietary drivers (just showed a black screen with (old?) Intel HD Graphics or nouveau).

The game was actually sometimes going too fast, I think the audio sped up too in response. I think tweaking VSync or the resolution settings might've been something that could trigger it (or stop it). No idea if FAudio would have any effect on this if it's even still an issue.

Also note the game is slow on computers with less resources even on Windows (but is still slightly slower on Wine)... it's strangely demanding. Not sure if it's caused by CPU or GPU, but I suspect it's GPU (unless it's Dencrapvo which caused Sonic Mania slowdowns in a recent update).

I expect to test it again with a recent AMD graphics card soon. It'll probably end up also testing how compatible it is with DXVK as this is a DX11 game (unless someone has already done that).

**Comment by aljelly** [comment]
Can confirm OP. Game seems to play fine at full speed except music doesn't play, and installing `xact` with winetricks fixes music.

**Comment by novadragonDOTspace** [comment]
With newest Proton Beta, audio plays flawlessly... but it still randomly freezes. The freezes seem more drastic than before.

**Comment by alemarciana** [comment]
Well, I found that even though audio/music plays with no problem on the latest proton beta, you still need to add the previously stated line on user.reg, that way, the crashes stop happening (at least that's how I got the crashes to stop)
> **Only tested on proton 3.16-4 beta!**
> 
> I got the music to work and fix crashing without **winetricks --force xact** by editing the games user.reg in steamapps/compatdata/546050/pfx/
> 
> Add new line under
> 
> [Software\Wine\DllOverrides]
> 
> `"xaudio2_7"="native,builtin"`

**Comment by alemarciana** [comment]
Hi, I have  trouble running Puyo Puyo Tetris, even though I got it working correctly (as stated on my comment above), the game suddenly stops launching, forcing me to disable Steam Play/Proton, re-enabling it, and re-downloading the game, the problem is that, just because one game stops working, I have to re-download all my games running under Steam Play, is there a way to disable steam play without losing game data?

**Comment by vjr2** [comment]
Si with the new faudio update this Runs out of the box now?

**Comment by robertonazareth** [comment]
The game doesn't even start, it crashes at the "this game uses autosave" screen at the very beggining.

**Comment by kisak-valve** [comment]
Hello, with Proton 4.2-1 (or 3.16-8), can someone please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

**Comment by vjr2** [comment]
ITS WORKING NOW!!!! ITS HAPPENING!!!! THANKS VALVE :D :D :D

**Comment by jolty1** [comment]
> Hello, with Proton 4.2-1 (or 3.16-8), can someone please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

[steam-546050.log](https://github.com/ValveSoftware/Proton/files/3022135/steam-546050.log)

**Comment by robertonazareth** [comment]
Ok, I said it wasn't working but i was going to reinstall to get the log and it... Worked! 

The game works perfectly now, thanks!

**Comment by jolty1** [comment]
> **Only tested on proton 3.16-4 beta!**
> 
> I got the music to work and fix crashing without **winetricks --force xact** by editing the games user.reg in steamapps/compatdata/546050/pfx/
> 
> Add new line under
> 
> [Software\Wine\DllOverrides]
> 
> `"xaudio2_7"="native,builtin"`

While the game runs and music does work since some proton versions ago.. You still need to edit the prefix as above to stop the game from random crashing.

**Comment by kisak-valve** [comment]
@jolty1, if possible, please collect a proton log of the random crash. The log should compress well in an archive if it's large.

**Comment by jolty1** [comment]
@kisak-valve 3 proton logs of crashes. The crashes can happen within 5-30+ minutes very unpredictable.

[steam-546050.tar.gz](https://github.com/ValveSoftware/Proton/files/3033387/steam-546050.tar.gz)

**Comment by flibitijibibo** [comment]
Seems it's not a crash and more of a deadlock:

https://github.com/FNA-XNA/FAudio/commit/0a7e525702cd21cfee39b9816a152a1ab25ae538

Try running this test binary:

```
cd ~/.local/share/Steam/SteamApps/common/Proton\ 4.2/dist/lib64
curl -O flibitijibibo.com/faudiogit64.tar.bz2
tar xvfj faudiogit64.tar.bz2
rm faudiogit64.tar.bz2
```

**Comment by jolty1** [comment]
Yep, that has fixed it. Played for a few hours without any deadlocks.

**Comment by vjr2** [comment]
Hi, game started to crash for me again. No matter what I do it will crash after 10 minutes of gameplay
[steam-546050.log](https://github.com/ValveSoftware/Proton/files/3050629/steam-546050.log)

**Comment by flibitijibibo** [comment]
@vjr2 See https://github.com/ValveSoftware/Proton/issues/359#issuecomment-479086467

**Comment by fmorency** [comment]
Game doesn't start for me. Tried using Proton 3.16, 4.2 and 4.11 without success. The game crashes after the first "Saving..." screen at the start of the game. I tried the various fixes mentioned in this thread without success.

[steam-546050.log](https://github.com/ValveSoftware/Proton/files/4017009/steam-546050.log)

**Comment by fmorency** [comment]
Same error as above with 4.11-12
[steam-546050.log](https://github.com/ValveSoftware/Proton/files/4082103/steam-546050.log)

**Comment by fmorency** [comment]
Using Proton 5.0-2 leads to a black screen + game freeze after the "Saving..." screen. Log file is 53Mb uncompressed.
[steam-546050.log.zip](https://github.com/ValveSoftware/Proton/files/4201687/steam-546050.log.zip)

**Comment by fmorency** [comment]
Game doesn't even start with Proton 5.0-4. No screen whatsoever. Log file is 52Mb uncompressed.
[steam-546050.log.zip](https://github.com/ValveSoftware/Proton/files/4319865/steam-546050.log.zip)

**Comment by RanAwaySuccessfully** [comment]
Game seems to work well with Proton 5.0-3 and Proton 4.11-13, but I had to set the launch options to `LANG=ja_JP.UTF-8 %command%` in order to get player's names to show up correctly on Online mode, otherwise CJK characters will not display.

Game does not work at all with Proton 5.0-4:

[steam-546050.log.zip](https://github.com/ValveSoftware/Proton/files/4324344/steam-546050.log.zip) (55MB uncompressed, 2.5MB compressed)

**Comment by kisak-valve** [comment]
Puyo Puyo Tetris (546050)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/3630.
@fagnerln posted on 2020-03-12T13:29:46:

# Compatibility Report
- Name of the game with compatibility issues: Puyo Puyo Tetris
- Steam AppID of the game: 546050

## System Information
- GPU: RX580
- Driver/LLVM version: Mesa 20.1
- Kernel version: 5.3


- Link to full system information report as [Gist](https://gist.github.com/fagnerln/18fa34bae26c4132b86936072dd83e10):
- Proton version: 5.0-4

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

I can't add the LOG because it's a very large file, with 561756 lines and 56MB, so I uploaded it to Google Drive:

https://drive.google.com/file/d/1Cc_aFGP14FFVKP_raldPquM8Z4Bdvswr/view?usp=sharing

## Symptoms 

The game simply don't run anymore, it worked perfectly fine until yesterday, with 5.0-3


## Reproduction

1- Click Play, and nothing happens

**Comment by aeikum** [comment]
Thanks for reporting the problem in 5.0-4. We have identified the problem and will have a fix in the next version. In the meantime you can use the "proton-5.0-3" branch for the Proton 5.0 tool by selecting it from the BETAS tab in tool's Properties dialog in the Steam client.

**Comment by deltadex7** [comment]
I have switched to Proton 5.0-3 and it still won't launch, even after waiting for like 2 hours. Log file is 56.7 MB. I have reinstalled both the game and Proton, but still no luck. Note that I was running on kernel 5.5.8. Will try again with 5.4 LTS kernel.
[steam-546050.log.zip](https://github.com/ValveSoftware/Proton/files/4328172/steam-546050.log.zip)

**Comment by kisak-valve** [comment]
Hello @deltadex7, `err:vulkan:wine_vk_instance_load_physical_devices Failed to enumerate physical devices, res=-3` `err:vulkan:wine_vkCreateInstance Failed to load physical devices, res=-3` in your log indicates that Vulkan failed to initialize on your system.

Please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by deltadex7** [comment]
Now that you have pointed that out, I proceed to install Vulkan on my system and restart Steam afterwards. The game finally runs as normal.

The [following gist](https://gist.github.com/deltadex7/0ae23c6ac6e11eeecd046b859e12189b) contains both system information before and after Vulkan install. The reason it failed before because I didn't have Vulkan for Intel before, only AMD/Radeon, which I don't have for graphics. And after installing the appropiate packages, there are no errors.

**Comment by aeikum** [comment]
Hello. Puyo Puyo Tetris should be playable again in Proton 5.0-5. Thanks for reporting the problem!

**Comment by fmorency** [comment]
> Hello. Puyo Puyo Tetris should be playable again in Proton 5.0-5. Thanks for reporting the problem!

Negative. Fresh install, Proton 5.0-5.
Game now starts but I still get the black screen + game freeze after the "Saving..." screen. Log attached.
[steam-546050.log.zip](https://github.com/ValveSoftware/Proton/files/4363467/steam-546050.log.zip)

**Comment by fagnerln** [comment]
> Hello. Puyo Puyo Tetris should be playable again in Proton 5.0-5. Thanks for reporting the problem!

It worked for me. Thanks!

**Comment by fmorency** [comment]
Still doesn't work here. Fresh install, Proton 5.0-6

Game starts but I get the black screen + game freeze after the "Saving..." screen. Log attached.

[steam-546050.log.zip](https://github.com/ValveSoftware/Proton/files/4486869/steam-546050.log.zip)

**Comment by fmorency** [comment]
This game works with Proton 5.0-7!!! Thank you!

**Comment by Vagankovo** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Puyo Puyo Tetris
- Steam AppID of the game: 546050

## System Information
- GPU: Radeon RX 5500M 4GB
- Driver/LLVM version: Mesa 21.2.2
- Kernel version: 5.14.7-2-MANJARO x86_64
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/Vagankovo/9cd2fbceb90029166ba122b6754c6afe
- Proton version: 6.13-7 and Experimental
- Desktop environment: GNOME on Wayland

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



## Symptoms 
- The game sometimes blacks out when switching workspaces. However, it still seems to be working, as I can move my mouse around and I can hear menu sounds.
- The game freezes in Swap matches

## Reproduction
Swap match freeze:
1. Load into a swap match.
2. Play until the game freezes.
> Note: sometimes the game doesn't freeze during the match. In this case, repeat.

Screen blackout:
1. Have the game launched.
2. Switch between workspaces and refocus into the game until the game blacks out.

## Logs
Swap match logs:
[steam-546050.log](https://github.com/ValveSoftware/Proton/files/7268143/steam-546050.log)
Screen blackout logs:
[steam-546050.log](https://github.com/ValveSoftware/Proton/files/7268175/steam-546050.log)

**Comment by Arborescent** [comment]
I experienced a massive slowdown in this game (~27 fps).
I used a profiler (sysprof) to understand what was consuming most of the time.

What I found is that the function `alx_get_stats` was being called thousands of times. That function is related to the network driver of a network card that I was not even using.
I disabled that ethernet card by using the command `sudo ip link set enp6s0 down`.

The result was an immediate correction of the FPS problem. Now the game exhibits perfect framerate.

tl;dr: If the game slows down, try disabling network cards you are not using.

**Comment by kisak-valve** [comment]
Puyo Puyo Tetris [546050]

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8194.
@cyo93 posted on 2024-10-26T04:54:17:

# Compatibility Report
- Name of the game with compatibility issues: Puyo Puyo Tetris
- Steam AppID of the game: 546050

## System Information
- GPU: 
- Video driver version: 24.0.5.183659.radeonsi_24.0.3-3
- Kernel version: 6.5.0-valve22-1-neptune-65-g9a338ed8a75e
- Link to full system information report as [Gist](https://gist.github.com/cyo93/486926ded91a23b918350de6c5325491):
- Proton version: 9.0-3

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



## Symptoms 
Since the Latest Steam Deck update I have trouble launching the game offline. Before it worked fine offline, but since the latest update, it sometimes doesn't work after a cold boot. 

## Reproduction
Start the game online, then try to run it offline. You should see it work and reach the SEGA screen. Then try to restart or shut down the Steam deck. Try launching the game again offline, and you may see the message in the video that stops you from playing the game offline.

[steam-546050.log](https://github.com/user-attachments/files/17529056/steam-546050.log) Before bug
[steam-546050_after.log](https://github.com/user-attachments/files/17529057/steam-546050_after.log) After bug
[systemreport-20241026042022.txt](https://github.com/user-attachments/files/17529059/systemreport-20241026042022.txt) System Report before bug
[systemreport-20241026042439.txt](https://github.com/user-attachments/files/17529061/systemreport-20241026042439.txt)
System Report after bug

Video: https://drive.google.com/file/d/1e92OvSfXdUs2jBsgMvfvm6oQq7TS65qn/view?usp=drive_link
